include/llbuild/Core/Clock.h

```diff
@@ -0,0 +1,39 @@
+//===- Clock.h --------------------------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLBUILD_CORE_CLOCK_h
+#define LLBUILD_CORE_CLOCK_h
+
+#include <cstdint>
+
+namespace llbuild {
+namespace core {
+
+class Clock {
```

Comment:
This should go in Basic, I think. I'm no expert on C++14+'s time functionality, but I wonder if we should just be using that directly?

Reply:
I'm inclined to suggest we use the standard library time functionality.

Reply:
Have you used it though? I seem to recall chatter that some people think it has serious problems

Reply:
I changed the implementation to use it. It looks much better as it handles conversion between units much better (or it does - in contrast to the previous implementation 🙂).
However: I don't see how it handles clock changes like traveling between timezones or getting the clock synchronized during a build (which is probably more likely).
I'll try to find out if it's possible to get monotonic values.

Reply:
`steady_clock` is monotonic and therefore does not get changed. I'll use that